data-generation/src/main/scala/thesis/SnapshotDelta.scala

```diff
@@ -167,53 +167,58 @@ object SnapshotDeltaObject {
+    val previousSnapshotEdgesKV = snapshot.edges.map(edge => (edge.attr._1, edge))
 
     val squashedEdgeActions = getSquashedActionsByEdgeId(logs)
 
     // Combine edges from the snapshot with the current log interval
     val joinedEdges = previousSnapshotEdgesKV.fullOuterJoin(squashedEdgeActions)
 
     // Create, update or omit (delete) edges based on values are present or not.
-    val newSnapshotEdges = joinedEdges.flatMap(outerJoinedEdge => outerJoinedEdge._2 match {
+    val newSnapshotEdges: RDD[Edge[(EdgeId, LTSV.Attributes)]] = joinedEdges.flatMap(outerJoinedEdge => outerJoinedEdge._2 match {
       case (None, None) => throw new IllegalStateException("The full outer join f*ucked up")
-      case (Some(previousEdge), None) => Some(Edge(outerJoinedEdge._1._1, outerJoinedEdge._1._2, previousEdge)) // Edge existed in the last snapshot, but no changes were done in this interval
+      case (Some(previousEdge), None) => Some(Edge(previousEdge.srcId, previousEdge.dstId, previousEdge.attr)) // Edge existed in the last snapshot, but no changes were done in this interval
       case (Some(previousEdge), Some(newEdgeAction)) => newEdgeAction.action match { // There has been a change to the edge in the interval
         case DELETE => None // Edge or {source,destination} vertex was deleted in the new interval
         case UPDATE => // Edge updated in this interval
-          Some(Edge(outerJoinedEdge._1._1, outerJoinedEdge._1._2, rightWayMergeHashMap(previousEdge, newEdgeAction.attributes)))
+          Some(previousEdge.copy(attr = (previousEdge.attr._1, rightWayMergeHashMap(previousEdge.attr._2, newEdgeAction.attributes))))
         case CREATE =>
           throw new IllegalStateException(s"$uuid An CREATE should never happen since this case should be an action referencing an existing entity")
       }
       case (None, Some(newEdgeAction)) => newEdgeAction.action match { // Edge was introduced in this interval
         case DELETE => None // Edge was introduced then promptly deleted (possibly due to a deleted vertex)
         case CREATE => // Edge was created
-          Some(Edge(outerJoinedEdge._1._1, outerJoinedEdge._1._2, newEdgeAction.attributes))
+          Some(logTSVToNewEdge(newEdgeAction))
         case UPDATE =>
           throw new IllegalStateException(s"$uuid An UPDATE should never happen if it's new because of how merging of LogTSVs are done $newEdgeAction")
       }
     })
     newSnapshotEdges
   }
 
-  // We are making the assumption that there can only be one edge between two nodes. We might have to create a surrogate key
-  def getSquashedActionsByEdgeId(logs: RDD[LogTSV]): RDD[((Long, Long), LogTSV)] = {
+  def logTSVToNewEdge(log: LogTSV): Edge[(EdgeId, Attributes)] = {
+    log.entity match {
+      case VERTEX(objId) => throw new IllegalStateException("This does not make sense. Only edges allowed")
+      case EDGE(id, srcId, dstId) => Edge(srcId, dstId, (id, log.attributes))
+    }
+  }
```

Comment:
```suggestion
  def logToEdge(log: LogTSV): Edge[(EdgeId, Attributes)] = {
    log.entity match {
      case VERTEX(objId) => throw new IllegalStateException("This does not make sense. Only edges allowed")
      case EDGE(id, srcId, dstId) => Edge(srcId, dstId, (id, log.attributes))
    }
  }
```
I og med at det ikke er en TSV, men et logobjekt